src/SFA.DAS.Commitments.Support.SubSite/Mappers/ApprenticeshipMapper.cs

```diff
@@ -0,0 +1,126 @@
+﻿using SFA.DAS.Commitments.Application.Queries.GetApprenticeshipsByUln;
+using SFA.DAS.Commitments.Support.SubSite.Extentions;
+using SFA.DAS.Commitments.Domain.Entities;
+using SFA.DAS.Commitments.Domain.Entities.DataLock;
+using SFA.DAS.Commitments.Support.SubSite.Extensions;
+using SFA.DAS.Commitments.Support.SubSite.Models;
+using SFA.DAS.HashingService;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Reflection;
+using System.Web;
+
+namespace SFA.DAS.Commitments.Support.SubSite.Mappers
+{
+    public class ApprenticeshipMapper : IApprenticeshipMapper
+    {
+        private readonly IHashingService _hashingService;
+
+        public ApprenticeshipMapper(IHashingService hashingService)
+        {
+            _hashingService = hashingService;
+        }
+
+        public UlnSummaryViewModel MapToUlnResultView(GetApprenticeshipsByUlnResponse response)
+        {
+            return new UlnSummaryViewModel
+            {
+                Uln = response.Apprenticeships.First().ULN,
+                ApprenticeshipsCount = response.TotalCount,
+                SearchResults = response.Apprenticeships.Select(o => MapToApprenticeshipSearchItemViewModel(o)).ToList()
+            };
+        }
+
+        public ApprenticeshipViewModel MapToApprenticeshipViewModel(Apprenticeship apprenticeship)
+        {
+            var changeRequested = apprenticeship.DataLockPriceTriaged || apprenticeship.DataLockCourseChangeTriaged;
+
+            return new ApprenticeshipViewModel
+            {
+                FirstName = apprenticeship.FirstName,
+                LastName = apprenticeship.LastName,
+                AgreementStatus = apprenticeship.AgreementStatus.GetEnumDescription(),
+                PaymentStatus = MapPaymentStatus(apprenticeship.PaymentStatus, apprenticeship.StartDate, apprenticeship.StopDate),
+                Alerts = MapRecordStatus(apprenticeship.UpdateOriginator, apprenticeship.DataLockCourseTriaged, changeRequested),
+                ULN = apprenticeship.ULN,
+                DateOfBirth = apprenticeship.DateOfBirth,
+                CohortReference = _hashingService.HashValue(apprenticeship.CommitmentId),
+                EmployerReference = apprenticeship.EmployerRef,
+                LegalEntity = apprenticeship.LegalEntityName,
+                TrainingProvider = apprenticeship.TrainingName,
+                UKPRN = apprenticeship.ProviderId,
+                Trainingcourse = apprenticeship.TrainingName,
+                ApprenticeshipCode = apprenticeship.TrainingCode,
+                DasTrainingStartDate = apprenticeship.StartDate,
+                DasTrainingEndDate = apprenticeship.StopDate,
```

Comment:
End Date <> StopDate
I'd expect we'd always want to show End Date, and show Stop Date when the apprenticeship is stopped (probably underneath the Stop status field, but that's a UI design decision).

Reply:
Will correct that